### PR TITLE
feat: update linking behaviour

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
in discussion with Jemma, this has updated routing to reflect updates to tickets https://digital.dclg.gov.uk/jira/browse/CLDC-1478 and https://digital.dclg.gov.uk/jira/browse/CLDC-1482

tested locally
minimal risk
no docs required

### CODE REVIEW

File: app/models/form/sales/pages/joint_purchase.rb [hunk 1]
```diff
@@ -5,6 +5,15 @@ def initialize(id, hsh, subsection)
...
     @subsection = subsection
+    @depends_on = [{
+      "ownershipsch" => 1,
+    },
+                   {
+                     "ownershipsch" => 2,
+                   },
+                   {
+                     "companybuy" => 2,
+                   }]
```
Comment (natdeanlewissoftwire) [inline app/models/form/sales/pages/joint_purchase.rb]: This spacing is required to pass linting - can I override rubocop somehow so I can fix this?

File: db/schema.rb [hunk 2]
```diff
@@ -348,15 +348,14 @@
     t.integer "age2"
     t.integer "age2_known"
     t.string "relat2"
-    t.string "otherrelat2"
```
Comment (natdeanlewissoftwire) [inline db/schema.rb]: Not sure why this was added back in as part of [this PR](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/commit/c52a30ae0c8cf335965340a1c03915d8068ee2a6), it shouldn't have been.

File: app/models/form/sales/pages/joint_purchase.rb (hunk 1 above)
Comment (ghost) [reply app/models/form/sales/pages/joint_purchase.rb]: I think if you put the hashes and their content on a single row rubocop won't complain, see https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/blob/main/app/models/form/lettings/questions/renewal.rb#L14

File: spec/models/form/sales/pages/joint_purchase_spec.rb [hunk 3]
```diff
@@ -26,4 +26,16 @@
   it "has the correct description" do
     expect(page.description).to eq("")
   end
+
+  it "has correct depends_on" do
+    expect(page.depends_on).to eq([{
+      "ownershipsch" => 1,
+    },
+                                   {
```
Comment (ghost) [inline spec/models/form/sales/pages/joint_purchase_spec.rb]: same as above

File: app/models/form/sales/pages/joint_purchase.rb (hunk 1 above)
Comment (natdeanlewissoftwire) [reply app/models/form/sales/pages/joint_purchase.rb]: Amazing, thanks!

**Review by CheshireSwift — APPROVED** [review]

**Pull request merged**